### DISCUSSION TITLE
coinv1 `is_account_registered` changes

### DISCUSSION THREAD

**Discussion by LawsonGraham** [aptos-labs/aptos-developer-discussions#272] [General]
I am seeing that unit tests which previously passed now are failing, and I believe I've traced this issue to here: https://github.com/aptos-labs/aptos-core/commit/d671bb8a594e4c428a0f00eea10c4adda7127f05#diff-8b0c87833e9d2dfab0fa028e93c4571213a9d059584ac8f1b71b41fd8d53803cL747

We depend on the mainnet branch, and I see that this breaking change was made to now abort rather than return false here... is this intended? and if so how should I modify code like this:

```
if (!coin::is_account_registered<AptosCoin>(signer::address_of(&caller))) {
            coin::register<AptosCoin>(&caller);
        }
```

to actually work then? my issue is coming from an external dependency...

Follow up:

I see that it errors given the AptosCoin store hasn't been created yet, however when I try to create this in my tests with `aptos_coin::initialize_for_test` I get the following error:

```
public entry fun create_pairing<CoinType>(
│     │                        -------------- In this function in 0x1::coin
│     ·  
│ 292 │ ╭         assert!(
│ 293 │ │             features::coin_to_fungible_asset_migration_feature_enabled(),
│ 294 │ │             error::invalid_state(EMIGRATION_FRAMEWORK_NOT_ENABLED)
│ 295 │ │         );
│     │ ╰─────────^ Test did not error as expected. Expected test to abort with code 104 originating in the module 092e95ed77b5ac815d3fbc2227e76db238339e9ca43ace45031ec2589bea5b8c::tiered_oracle but instead it aborted with code 196634 originating in the module 0000000000000000000000000000000000000000000000000000000000000001::coin rooted here
```
which seems to be coming from the FA migration being in progress... is there an easier way to get around this? else it feels like I'm kind of stuck here

**Comment by 0xbe1** [comment]
seems this was caught earlier in code review but it was somehow ignored https://github.com/aptos-labs/aptos-core/commit/d671bb8a594e4c428a0f00eea10c4adda7127f05#r143155278

**Reply by kshitijc1506** [reply]
Triaged this internally, thanks for flagging. Looking into it.

**Comment by lightmark** [comment]
@LawsonGraham Hi, thanks for flagging out. For your test rn, the quick fix is to compile your aptos cli from main branch. After release new CLI, you don't have to create the pairing.

**Reply by 0xbe1** [reply]
> @LawsonGraham Hi, thanks for flagging out. For your test rn, the quick fix is to compile your aptos cli from main branch. After release new CLI, you don't have to create the pairing.

Hey @lightmark do you know the timeline for new CLI release?

**Comment by 0xbe1** [comment]
a work around is to enable the said feature flag `coin_to_fungible_asset_migration_feature` in unit tests

```move
features::change_feature_flags_for_testing(aptos_framework, vector[features::get_coin_to_fungible_asset_migration_feature()], vector[]);
let (burn_cap, mint_cap) = aptos_coin::initialize_for_test(aptos_framework);
coin::destroy_burn_cap(burn_cap);
coin::destroy_mint_cap(mint_cap);
```
